File: lexilearner/models/ReadingContent.ts
export interface ReadingContent {
  Id: string;
  Title: string;
  Author?: string;
  Description: string;
  Cover: string;
  Content: string;
  Genre: string;
  Difficulty: number;
}

export interface ReadingItem {
  id: string;
  type: string;
  title: string;
  author?: string;
  description?: string;
  cover: string;
  content: string;
  genre: string;
  difficulty: number;
}

File: server/Models/ReadingContent.cs
using System.ComponentModel.DataAnnotations.Schema;
namespace LexiLearner.Models
{
    public class ReadingContent
    {
        public Guid Id { get; set; } = Guid.NewGuid();

        public required string Title { get; set; }
        public required string Cover { get; set; }
        public required string Content { get; set; }

        public required Guid GenreId { get; set; }
        [ForeignKey("GenreId")]
        public required Genre Genre { get; set; }

        public required float Difficulty { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }
}
